client/src/util/dataframe/labelIndex.ts

```diff
@@ -214,82 +237,70 @@ class DenseInt32Index {
     }
 
     this.minLabel = minLabel;
-    this.rindex = labels;
+    this.rindex = int32Labels;
     this.index = index;
-    this.__id = __getMemoId();
-    this.__compile();
-  }
-
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  __compile() {
-    const { minLabel, index, rindex } = this;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getOffset = function getOffset(l: any) {
-      if (!Number.isInteger(l)) return undefined;
-      const offset = index[l - minLabel];
-      return offset === -1 ? undefined : offset;
-    };
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getOffsets = function getOffsets(arr: any) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      return arr.map((i: any) => this.getOffset(i));
-    };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getLabel = function getLabel(i: any) {
-      return Number.isInteger(i) ? rindex[i] : undefined;
+    this.getOffset = function getOffset(label: LabelType) {
+      if (!Number.isInteger(label)) return -1;
+      const lblIdx: number = <number>label - minLabel;
+      if (lblIdx < 0 || lblIdx >= index.length) return -1;
+      const offset = index[lblIdx];
+      return offset;
     };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getLabels = function getLabels(arr: any) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      return arr.map((i: any) => this.getLabel(i));
+    this.getLabel = function getLabel(offset: number) {
+      return Number.isInteger(offset) ? labels[offset] : undefined;
```

Comment:
same question here for `undefined` vs. `-1`

Reply:
same answer as above.  See the abstract method definition.